tracer/src/Datadog.Trace/DatabaseMonitoring/DatabaseMonitoringPropagator.cs

```diff
@@ -133,12 +135,34 @@ internal static bool PropagateDataViaContext(DbmPropagationLevel propagationLeve
                 parameter.DbType = DbType.Binary;
                 injectionCommand.Parameters.Add(parameter);
 
-                injectionCommand.ExecuteNonQuery();
-
                 if (Log.IsEnabled(LogEventLevel.Debug))
                 {
                     // avoid building the string representation in the general case where debug is disabled
-                    Log.Debug("Span data for DBM propagated for {Integration} via context_info with value {ContextValue} (propagation level: {PropagationLevel}", integrationId, HexConverter.ToString(contextValue), propagationLevel);
+                    Log.Debug("Propagating span data for DBM for {Integration} via context_info with value {ContextValue} (propagation level: {PropagationLevel}", integrationId, HexConverter.ToString(contextValue), propagationLevel);
+                }
+
+                try
+                {
+                    injectionCommand.ExecuteNonQuery();
+                }
+                catch (Exception e)
+                {
+                    if (_remainingErrorLogs > 0) // stop logging the error after a while
+                    {
+                        var actualRemaining = Interlocked.Decrement(ref _remainingErrorLogs);
+                        if (actualRemaining >= 0)
+                        {
+                            var message = "Error setting context_info [{ContextValue}] for DB query, falling back to service only propagation mode. There won't be any link with APM traces.";
+                            if (actualRemaining == 0)
+                            {
+                                message += " Will stop logging this error now.";
```

Comment:
[non-blocking] Note that this causes the log template to change for the last entry. Consider instead using a single template and including either the number of retries performed (counting up from 0 to 100) or the number of retries left (counting down from 100 to 0).